test/dhcp/subnet_service_test.rb

```diff
@@ -49,6 +49,59 @@ def test_find_subnet
     assert_equal subnets.first, @service.find_subnet("192.168.0.0")
   end
 
+  def test_find_subnet_with_cidr_mask
+    subnets = [Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.64", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.128", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.192", "255.255.255.192")]
+    @service.add_subnets(*subnets)
+    assert_equal subnets[2], @service.find_subnet("192.168.0.131")
+  end
+
+  def test_find_subnet_with_cidr_mask_and_match_in_a_group_of_one_network
+    subnets = [Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.64", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.192", "255.255.255.192")]
+    @service.add_subnets(*subnets)
+    assert_equal subnets[0], @service.find_subnet("192.168.0.10")
+  end
+
+  def test_find_subnet_with_cidr_mask_if_subnet_is_undefined
+    subnets = [Proxy::DHCP::Subnet.new("192.168.0.0", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.64", "255.255.255.192"),
+               Proxy::DHCP::Subnet.new("192.168.0.192", "255.255.255.192")]
+    @service.add_subnets(*subnets)
+    assert_nil @service.find_subnet("192.168.0.131")
+  end
+
+  # the address we are looking for is in a /27 subnet
+  def test_find_subnet_with_mixed_cidr_masks_deep
```

Comment:
These two names aren't great (esp. seeing that you added comments here). Perhaps something like ```test_find_subnet_with_host_in_255_255_255_224_subnet``` or something like that?